pkg/indexgateway/gateway.go

```diff
@@ -225,16 +226,12 @@ func (g *Gateway) GetChunkRef(ctx context.Context, req *logproto.GetChunkRefRequ
 		return nil, err
 	}
 
-	series := make(map[uint64]labels.Labels)
 	result = &logproto.GetChunkRefResponse{
 		Refs: make([]*logproto.ChunkRef, 0, len(chunks)),
 	}
 	for _, cs := range chunks {
 		for i := range cs {
 			result.Refs = append(result.Refs, &cs[i].ChunkRef)
-			if _, ok := series[cs[i].Fingerprint]; !ok {
-				series[cs[i].Fingerprint] = cs[i].Metric
```

Comment:
nit: What about adding a comment in the chunk struct about fields not being populated until fetched?

Reply:
^ I would also like this, it seems inevitable that someone is going to run into this again since it's not very obvious.